pkg/client/unversioned/clientcmd/loader_test.go

```diff
@@ -224,14 +224,15 @@ func TestResolveRelativePaths(t *testing.T) {
 	}
 
 	configDir1, _ := ioutil.TempDir("", "")
+	defer os.RemoveAll(configDir1)
 	configFile1 := path.Join(configDir1, ".kubeconfig")
 	configDir1, _ = filepath.Abs(configDir1)
-	defer os.Remove(configFile1)
+
 	configDir2, _ := ioutil.TempDir("", "")
+	defer os.RemoveAll(configDir2)
```

Comment:
NB: This is subtle but works.  I had to look it up :)  Because configDir2 gets rewritten below, we are relying on the fact that defer will capture the referenced variables. https://play.golang.org/p/5cBaEM9wgJ